src/plugin-store-item.svelte

```diff
@@ -0,0 +1,226 @@
+<script lang="ts">
+import Menu from "@smui/menu";
+import { Button, SplitButton } from "./components/button";
+import List, { Item, Text } from "@smui/list";
+
+type PluginKind = "editor" | "menu" | "validator";
+const menuPosition = ["top", "middle", "bottom"] as const;
+type MenuPosition = (typeof menuPosition)[number];
+
+type Plugin = {
+	name: string;
+	author?: string;
+	src: string;
+	icon?: string;
+	kind: PluginKind;
+	requireDoc?: boolean;
+	position?: MenuPosition;
+	installed: boolean;
+	official?: boolean;
+	description?: string;
+};
+
+type ConfigurePluginDetail = {
+	name: string;
+	// The API describes only 'menu' and 'editor' kinds
+	// but we still use the 'validator' too, so I just use the type PluginKind
+	kind: PluginKind;
+	config: Plugin | null;
+};
+
+export let filteredPlugins: Plugin[];
+export let pluginStore: Element;
+export let plugin: Plugin;
+export let index: number;
+export let storedPlugins: Plugin[];
+export let localPlugins: Plugin[];
+
+function dispatchConfigurePlugin(plugin: Plugin, shouldDelete = false) {
+	const event = new CustomEvent<ConfigurePluginDetail>(
+		"oscd-configure-plugin",
+		{
+			bubbles: true,
+			composed: true,
+			detail: {
+				name: plugin.name,
+				kind: plugin.kind,
+				config: shouldDelete ? null : plugin,
+			},
+		},
+	);
+
+	pluginStore.dispatchEvent(event);
+}
+
+function installExternalPlugin(plugin: Plugin) {
+	const currentPlugins = storedPlugins;
+
+	const pluginCopy = { ...plugin };
+	pluginCopy.installed = true;
+	currentPlugins.push(pluginCopy);
+
+	localPlugins = currentPlugins;
+
+	dispatchConfigurePlugin(pluginCopy);
+	console.log("Installed external plugin:", plugin.name);
+}
+
+// Completely removes plugin from local browser cache.
+function uninstallExternalPlugin(plugin: Plugin) {
+	const currentPlugins = storedPlugins;
+	const updatedPlugins = currentPlugins.filter((it) => it.name !== plugin.name);
+	localPlugins = updatedPlugins;
+
+	dispatchConfigurePlugin(plugin, true);
+	console.log("Uninstalled external plugin:", plugin.name);
+}
+
+// Enables/disables plugin by toggling the "installed" property.
+function toggleOfficialPlugin(plugin: Plugin, isEnabled: boolean) {
+	const currentPlugins = storedPlugins;
+	const foundPlugin = currentPlugins.find((it) => it.name === plugin.name);
+	if (foundPlugin) {
+		foundPlugin.installed = isEnabled;
+	}
+
+	localPlugins = currentPlugins;
+	plugin.installed = isEnabled;
+
+	dispatchConfigurePlugin(plugin);
+	console.log("Set toggle state for", plugin.name);
+}
+
+let menus: Menu[];
+$: menus = filteredPlugins.map(() => null);
+$: menuStates = filteredPlugins.map(() => false);
+
+function openPluginMenu(index: number) {
+	menuStates = menuStates.map(() => false);
+	menuStates[index] = true;
+}
+
+function alternateRowColors(index: number) {
+	return `background: rgba(0, 0, 0, ${index % 2 === 0 ? "0" : "0.03"});`;
+}
```

Comment:
We usually can solve this with CSS. Is there reason why we do this with JavaScript?

Reply:
That was just something I was familiar with, I can move it to CSS though